### PR TITLE
Add user approval error codes

### PR DESCRIPTION
**Pull request by ayshsandu**

### CODE REVIEW

File: en/docs/references/error-catalog.md [hunk 1]
```diff
@@ -216,3 +216,78 @@ This document describes all the REST API error codes that are used in WSO2 Ident
...
+        <td>HTA-10002</a></td>
+        <td>400</td>
+        <td>Invalid Input Provided.</td>
+        <td>The provided Task ID is not in the correct format.</td>
+      </tr>
+      <tr>
+        <td>HTA-10003</a></td>
+        <td>404</td>
+        <td>Task does not exists.</td>
+        <td> </td>
```
Comment (sherenewso2) [inline en/docs/references/error-catalog.md]: Can we change this to "Task does not **exist**"?

File: en/docs/references/error-catalog.md [hunk 2]
```diff
@@ -216,3 +216,78 @@ This document describes all the REST API error codes that are used in WSO2 Ident
...
+        <td>400</td>
+        <td>Unable to update the approval status.</td>
+        <td>Invalid state change is requested for the given task.</td>
+      </tr>
+      <tr>
+        <td>HTA-15002</a></td>
+        <td>500</td>
+        <td>Unable to retrieve approvals for the user.</td>
+        <td>Server Encountered an error while retrieving approvals for user.</td>
+      </tr>
```
Comment (sherenewso2) [inline en/docs/references/error-catalog.md]: can we make the "e" in "Encountered" simple case? Same for the next three error codes as well please.

File: en/docs/references/error-catalog.md [hunk 3]
```diff
@@ -216,3 +216,78 @@ This document describes all the REST API error codes that are used in WSO2 Ident
...
+    <th>HTTP Status Code</th>
+    <th>Error Message</th>
+    <th>Possible Cause</th>
+  </thead>
+  <tbody>
+      <tr>
+        <td>HTA-10001</a></td>
+        <td>403</td>
+        <td>Access Denied.</td>
+        <td>You are not authorized to perform this task.</td>
```
Comment (sherenewso2) [inline en/docs/references/error-catalog.md]: Only the first letter of the first word of the error message needs to be a capital letter. The rest of the words in the error message can start with simple letters. So in this case, can we change this to "Access denied" and the next one to "Invalid input provided" please?

File: en/docs/references/error-catalog.md (hunk 2 above)
Comment (ayshsandu) [reply en/docs/references/error-catalog.md]: updated

File: en/docs/references/error-catalog.md (hunk 3 above)
Comment (ayshsandu) [reply en/docs/references/error-catalog.md]: updated

**Pull request merged**